### ISSUE TITLE
Finetune recipe for classification on custom dataset?

### ISSUE THREAD

**Issue by revantteotia** [McGill-NLP/llm2vec#28]
Great work btw!!

I was wondering if you have a finetuning recipe for classification on custom dataset?

I am working on one using the codebase. would be happy to raise a PR if it works well.

Thank you!

**Comment by vaibhavad** [comment]
Do you mean using trained LLM2Vec model for classification? We have included an example for that - https://github.com/McGill-NLP/llm2vec/blob/main/examples/classification.py

**Comment by revantteotia** [comment]
Hi Vaibhav,

The example is a great explanation of using LLM2VEC to encode text for classification.
But as far as I could figure out the code, the `.encode()` does not propagate gradients back to the model weights.

I want to further finetune the model weights for our custom dataset. 

I could not find any example of further finetune the model weights for classification. Anyway, I am working on it :)

**Comment by vaibhavad** [comment]
Hi @revantteotia,

We are in the process of releasing the training code. You can check the WIP in [supervised-training](https://github.com/McGill-NLP/llm2vec/tree/supervised-training) branch. The code is functional at this point but I need to add proper documentation and steps.

**Comment by monilouise** [comment]
Hi @vaibhavad ,

I think I have a question similar to @revantteotia.  In the examples, you train a logistic regression from a model pre-trained on MNTP with supervised contrastive learning.  However, you assume the logistic regression will be responsible for the final classification task; you don't assume a fine-tuning step on the target task.  Is it correct?

**Comment by vaibhavad** [comment]
hi @monilouise,

That is correct. The setting is consistent with how [MTEB](https://github.com/embeddings-benchmark/mteb) benchmark evaluates a model's capability to be a universal sentence encoder. However, if you are concerned only with a specific classification task and if sufficient training data is available, I believe fine-tuning the logistic regression and model together will perform better.
